settings.h

```diff
@@ -25,8 +25,13 @@
 #include "system.h"
 
 
-// define GRBL version and build
-#define GRBL_VERSION "0.9K.3.7"
+// This is a bit of compiler magic to ensure that we don't
+// accidentally optimize away variables that are only used by external
+// tools (version numbers, ident blocks, etc)
+#define ALWAYS_KEEP(v) __asm__ __volatile__("" :: "m" (v))
+
+#define STR(s) #s
+#define XSTR(s) STR(s)
```

Comment:
It's a preprocessor trick.  STR will simply paste in the token, so

`"VERSION=" STR(GRBL_VERSION)` would evaluate to `VERSION=GRBL_VERSION`

By wrapping the initial token, it forces the preprocessor to evaluate the token first, so it pastes in the actual string content of the definition.  It's a very commonly used trick in firmware.